impl/src/main/java/org/jboss/weld/xml/BeansXmlParser.java

```diff
@@ -59,10 +60,21 @@
  */
 public class BeansXmlParser {
 
+    private static final SAXParserFactory PARSER_FACTORY = SAXParserFactory.newInstance();
+
     private static final InputSource[] EMPTY_INPUT_SOURCE_ARRAY = new InputSource[0];
 
+    private static final int BEANS_XML_BUFFER_SIZE = 1024;
+
+    static {
+        PARSER_FACTORY.setValidating(!SystemPropertiesConfiguration.INSTANCE.isXmlValidationDisabled());
+        PARSER_FACTORY.setNamespaceAware(true);
+    }
+
     private Function<URL, BeansXml> URL_TO_BEANS_XML_FUNCTION = BeansXmlParser.this::parse;
 
+    private SAXParser parser;
```

Comment:
I'm afraid we cannot share - `SAXParser` is not guaranteed to be thread-safe and `BeansXmlParser.parse(URL)` can be used concurrently.

Reply:
OK.  I was going off the class documentation, which reads: "This class is NOT threadsafe, and should only be called in a single thread".  Indeed, if this documentation is incorrect and the class is called from multiple threads, then sharing the parser won't work.

Reply:
BTW `BeansXmlParser` itself is (incorrectly) marked as NOT threadsafe.